srcts/src/components/_shinyRemovedObserver.ts

```diff
@@ -0,0 +1,106 @@
+type Callback<T> = (el: T) => void;
+
+/**
+ * Watch for the removal of specific elements from regions of the page.
+ */
+export class ShinyRemovedObserver {
+  private observer: MutationObserver;
+  private watching: Set<HTMLElement>;
+
+  /**
+   * Creates a new instance of the `ShinyRemovedObserver` class to watch for the
+   * removal of specific elements from part of the DOM.
+   *
+   * @param selector A CSS selector to identify elements to watch for removal.
+   * @param callback The function to be called on a matching element when it
+   * is removed.
+   */
+  constructor(selector: string, callback: Callback<HTMLElement>) {
+    this.watching = new Set<HTMLElement>();
+    this.observer = new MutationObserver((mutations) => {
+      const found = [];
```

Comment:
I think `found` should also be a `Set` so we're not executing a callback several times for the same element?

Reply:
Yeah a `Set()` is a good idea to make sure we don't duplicate work. That _does_ require `found`. And wrapping in `try` is also a good idea to make sure that `callback()` is called on all found elements, even if one of them fails.